main_tests/tests_main.c

```diff
@@ -0,0 +1,214 @@
+}
+// a verifier !!!!!
+void test_pop(void)
+{
+    linked_list *test_pop = new_element(5);
+    int liste_value[] = {12, 25, 9, 45, 4};
+    for (int i = 0; i < 5; i++)
+    {
+        push(test_pop, liste_value[i]);
+    }
+    CU_ASSERT(length(test_pop) == 6);
+    linked_list *pt = test_pop;
+    CU_ASSERT(pt->data == 5);
+    for (int i = 0; i < 5; i++)
+    {
+        pt = pt->next;
+        CU_ASSERT(pt->data == liste_value[i]);
+    }
+    pt = pt->previous;
+    pop(test_pop);
+    CU_ASSERT(length(test_pop) == 5);
+
+    CU_ASSERT(pt->data == 45);
+    pt = pt->previous;
+    pop(test_pop);
+    CU_ASSERT(length(test_pop) == 4);
+
+    CU_ASSERT(pt->data == 9);
+    list_free(test_pop);
+}
+
+void test_add_index(void)
+{
+    linked_list *test_add_index = new_element(10);
+    linked_list *pt = test_add_index;
+    int list_value[] = {20, 40, 50, 60, 70};
+    int new_list_value[] = {20, 30, 40, 50, 60, 70};
+    for (int i = 0; i < 5; i++)
+    {
+        push(test_add_index, list_value[i]);
+    }
+    CU_ASSERT(length(test_add_index) == 6);
+    add_index(test_add_index, 30, 3);
+    CU_ASSERT(length(test_add_index) == 7);
+    pt = pt->next;
+    for (int i = 0; i < 5; i++)
+    {
+        CU_ASSERT(pt->data == new_list_value[i]);
+        pt = pt->next;
+    }
+    list_free(test_add_index);
+}
+
+void test_add_before(void)
+{
+    linked_list *test_add_before = new_element(10);
+    linked_list *pt = test_add_before;
+    int liste_value[] = {20, 30, 40, 50, 60, 70};
+    int new_list_value[] = {20, 30, 40, 45, 50, 60, 70};
+    for (int i = 0; i < 6; i++)
+    {
+        push(test_add_before, liste_value[i]);
+    }
+    CU_ASSERT(length(test_add_before) == 7);
+    add_before(test_add_before, 45, 50);
+    CU_ASSERT(length(test_add_before) == 8);
+    for (int i = 0; i < 7; i++)
+    {
+        pt = pt->next;
+        CU_ASSERT(pt->data == new_list_value[i]);
+    };
+    list_free(test_add_before);
+}
+
+void test_add_after(void)
+{
+    linked_list *test_add_after = new_element(10);
+    linked_list *pt = test_add_after;
+    int list_value[] = {20, 30, 40, 50, 60, 70};
+    int new_list_value[] = {20, 30, 40, 45, 50, 60, 70};
+    for (int i = 0; i < 6; i++)
+    {
+        push(test_add_after, list_value[i]);
+    }
+    CU_ASSERT(length(test_add_after) == 7);
+    add_after(test_add_after, 45, 40);
+    CU_ASSERT(length(test_add_after) == 8);
+    for (int i = 0; i < 7; i++)
+    {
+        pt = pt->next;
+        CU_ASSERT(pt->data == new_list_value[i]);
+    };
+    list_free(test_add_after);
+}
+
+void test_revert(void)
+{
+    linked_list *test_revert = new_element(10);
+    linked_list *pt = test_revert;
+    int list_value[] = {20, 30, 40, 50, 60, 70};
+    // int new_list_value[] = {70, 60, 50, 40, 30, 20, 10};
+    for (int i = 0; i < 6; i++)
+    {
+        push(test_revert, list_value[i]);
+    }
+    CU_ASSERT(length(test_revert) == 7);
+    CU_ASSERT(pt->data == 10);
+    for (int i = 0; i < 6; i++)

```

Comment:
je pense que tu en es là, quand ce sera ok, tu pourra simplement utiliser une boucle dans l'autre sens :
for (int i = 6; i >= 0; i--)
``